source/slang/slang-lower-to-ir.cpp

```diff
@@ -8417,6 +8473,16 @@ RefPtr<IRModule> generateIRForTranslationUnit(
 #endif
 
     validateIRModuleIfEnabled(compileRequest, module);
+    
+    // Process higher-order-function calls before any optimization passes
+    // to allow the optimizations to affect the generated funcitons.
+    // 1. Process JVP derivative functions.
+    processJVPDerivativeMarkers(module);
```

Comment:
A better way is to inspect all calls first, then generate a diff func for the callee if it hasn't been generated and replace the callee inst with the generated func. This way we don't do work for funcs that are not actually called, and we don't need to store a reference in the marker decoration.